bson/src/main/org/bson/AbstractBsonWriter.java

```diff
@@ -750,39 +752,53 @@ public void close() {
 
     @Override
     public void pipe(final BsonReader reader) {
-        pipeDocument(reader);
+        notNull("reader", reader);
+        pipeDocument(reader, null);
     }
 
-    private void pipeDocument(final BsonReader reader) {
```

Comment:
I probably would kept this and called the new method with `pipeDocument(reader, null);` as less changes are required.